bitnami/cilium/templates/configmap.yaml

```diff
@@ -0,0 +1,136 @@
+# Operator configuration
+operator-api-serve-addr: {{ printf ":%d" (int .Values.operator.containerPorts.api) | quote }}
+disable-endpoint-crd: "false"
+skip-crd-creation: "false"
+identity-allocation-mode: crd
+ipam: "cluster-pool"
+cluster-pool-ipv4-cidr: "10.0.0.0/8"
+cluster-pool-ipv4-mask-size: "24"
+# Hubble configuration
+enable-hubble: "true"
+hubble-socket-path: "/opt/bitnami/cilium/var/run/hubble.sock"
+hubble-export-file-max-size-mb: "10"
+hubble-export-file-max-backups: "5"
+hubble-listen-address: {{ printf ":%d" (int .Values.agent.containerPorts.hubblePeer) | quote }}
+hubble-disable-tls: {{ ternary "false" "true" .Values.tls.enabled | quote }}
+{{- if .Values.tls.enabled }}
+hubble-tls-cert-file: /certs/hubble/tls.crt
+hubble-tls-key-file: /certs/hubble/tls.key
+hubble-tls-client-ca-files: /certs/hubble/ca.crt
+{{- end }}
+{{- if or .Values.agent.metrics.enabled .Values.agent.hubbleMetrics.enabled .Values.operator.metrics.enabled }}
+# Prometheus metrics
+enable-metrics: "true"
+{{- if .Values.agent.metrics.enabled }}
+prometheus-serve-addr: {{ printf ":%d" (int .Values.agent.containerPorts.metrics) | quote }}
+controllerGroupMetrics: all
+metrics: ~
+{{- end }}
+{{- if .Values.agent.hubbleMetrics.enabled }}
+hubble-metrics-server: {{ printf ":%d" (int .Values.agent.containerPorts.hubbleMetrics) | quote }}
+hubble-metrics: ~
+{{- end }}
+{{- if .Values.operator.metrics.enabled }}
+operator-prometheus-serve-addr: {{ printf ":%d" (int .Values.operator.containerPorts.metrics) | quote }}
+{{- end }}
+{{- if .Values.envoy.metrics.enabled }}
+proxy-prometheus-port: {{ printf "%d" (int .Values.envoy.containerPorts.metrics) | quote }}
+{{- end }}
+{{- end }}
+# Envoy configuration
+external-envoy-proxy: "true"
+disable-envoy-version-check: "true"
+# Other configuration
+enable-k8s-networkpolicy: "true"
+synchronize-k8s-nodes: "true"
+remove-cilium-node-taints: "true"
+set-cilium-node-taints: "true"
+set-cilium-is-up-condition: "true"
+agent-not-ready-taint-key: "node.cilium.io/agent-not-ready"
+{{- end }}
+{{- end }}
+
+{{- if not .Values.existingConfigmap }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "common.names.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/part-of: cilium
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+{{- $configuration := include "cilium.configuration" . | fromYaml -}}
+{{- if .Values.overrideConfiguration }}
+{{- $overrideConfiguration := include "common.tplvalues.render" (dict "value" .Values.overrideConfiguration "context" .) | fromYaml }}
+{{- $configuration = mergeOverwrite $configuration $overrideConfiguration }}
```

Comment:
I would use `mustMergeOverwrite` so it fails in case of unsuccessful merge

Reply:
addressed at https://github.com/bitnami/charts/pull/26374/commits/d6b150377935aae125234d5476f88708f3d93c5e